File: Frontend/cashflow/src/types/BankAccount.ts
export class BankAccount {
    id: number;
    type: number;
    name: string;
    balance: number;
    creditBalance: number;
    createdAt: string;
    updatedAt: string;

    constructor(
        id: number,
        type: number,
        name: string,
        balance: number,
        creditBalance: number,
        createdAt: string,
        updatedAt: string
    ) {
        this.id = id;
        this.type = type;
        this.name = name;
        this.balance = balance;
        this.creditBalance = creditBalance;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }
}

File: Backend/Models/BankAccount.cs
﻿using System.ComponentModel.DataAnnotations.Schema;

namespace CashFlow.Models;

public class BankAccount : Entity
{
    public int Id { get; set; }
    [Column(TypeName = "nvarchar(50)")]
    public BankAccountType Type { get; set; }

    public string Name { get; set; } = string.Empty;// Name should be created as user surname+bankaccountid+type
    [Column(TypeName = "decimal(18, 2)")]
    public double Balance { get; set; }
    [Column(TypeName = "decimal(18, 2)")]
    public double CreditBalance { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public BankAccount()
    {
        CreatedAt = DateTime.Now;
        UpdatedAt = CreatedAt;
    }
    public int UserId { get; set; }
    public User? User { get; set; }
}